packages/react-components/src/components/datatree-output-component.tsx

```diff
@@ -188,4 +190,51 @@ export class DataTreeOutputComponent extends AbstractOutputComponent<AbstractOut
             this._debouncedFetchSelectionData();
         }
     }
+
+    private async exportOutput(): Promise<void> {
+        const focusContainer = document.getElementById(this.props.traceId + this.props.outputDescriptor.id + 'focusContainer');
+        if (focusContainer) {
+            const table = focusContainer.querySelector('div:nth-child(2) > table');
+            if (table) {
+                const rows = table.querySelectorAll('tr');
+
+                const csvArray = [];
+                for (let i = 0; i < rows.length; i++) {
+                    const row = [];
+                    const cols = rows[i].querySelectorAll('td, th');
+                    for (let j = 0; j < cols.length - 1; j++) {
+                        let data;
+                        const content = cols[j].textContent;
+                        if (content) {
+                            data = content.replace(/\s*\|/g, '');
+                        } else {
+                            data = content;
+                        }
+                        row.push(data);
+                    }
+                    csvArray.push(row.join(','));
+                }
+                const tableString = csvArray.join('\n');
+
+                const link = document.createElement('a');
+                link.setAttribute('href', `data:text/csv;charset=utf-8,${encodeURIComponent(tableString)}`);
+                link.setAttribute('download', (this.props.traceName ?? 'export') + ' - ' + this.props.outputDescriptor.name);
+
+                link.style.display = 'none';
+                document.body.appendChild(link);
+
+                link.click();
+
+                document.body.removeChild(link);
+            }
+        }
+    }
+
+    protected showAdditionalOptions(): React.ReactNode {
+        return <React.Fragment>
+            <ul>
+                <li className='drop-down-list-item' key={0} onClick={() => this.exportOutput()}>Export to csv</li>
```

Comment:
The menus in the output components are currently hard-coded because there are only a few options. But more and more will be added soon and we should have a generic way to provide menu and commands. The advantage will be to decouple the implementation from the output classes itself as well as it easier to customize or change commands. Please backlog this task.

The menu has a different look and feel than the rest of the application. It should be possible the css style it so that we can re-use the theia styles for it.